src/other/tables.jl

```diff
@@ -21,7 +21,7 @@ end
 Tables.columnindex(df::Union{AbstractDataFrame, DataFrameRow}, idx::AbstractString) =
     columnindex(df, Symbol(idx))
 
-Tables.schema(df::AbstractDataFrame) = Tables.Schema(propertynames(df), eltype.(eachcol(df)))
+Tables.schema(df::AbstractDataFrame) = Tables.Schema{Tuple(_names(df)), Tuple{[eltype(col) for col in eachcol(df)]...}}()
```

Comment:
I'm a bit late, but wouldn't a generator be even faster?
```julia
Tuple{(eltype(col) for col in eachcol(df))...}
```

Reply:
It seems it would be slower (I think I already checked it back then):
```
julia> f(y) = Tuple{[x for x in y]}
f (generic function with 1 method)

julia> g(y) = Tuple{(x for x in y)...}
g (generic function with 1 method)

julia> @time f(1:5);
  0.000007 seconds (4 allocations: 304 bytes)

julia> @time g(1:5);
  0.000012 seconds (8 allocations: 336 bytes)

julia> @time f(1:100);
  0.000011 seconds (4 allocations: 1.781 KiB)

julia> @time g(1:100);
  0.000017 seconds (106 allocations: 6.594 KiB)

julia> @time f(1:1000);
  0.000076 seconds (494 allocations: 31.562 KiB)

julia> @time g(1:1000);
  0.000115 seconds (1.99 k allocations: 85.375 KiB)
```
(and in general - probably it is a minimal overhead I think)